### PR TITLE
parser pull: Add support for reusing parser

### PR DESCRIPTION
**Pull request by DmitryPogrebnoy**
GitHub: Fix GH-214 

This is for parsing XML documents stream. We can use one parser to parse multiple XML documents with this feature.

### CODE REVIEW

**Comment by kou** [comment]
@naitoh Do you want to review this?

**Comment by naitoh** [comment]
> Do you want to review this?

Looks good to me.

File: test/test_pullparser.rb [hunk 1]
```diff
@@ -156,6 +156,43 @@ def test_peek
...
+      xml_chunks.each do |chunk|
+        writer.write(chunk)
+      end
+      writer.close
+
+      parser = REXML::Parsers::PullParser.new(reader)
+      messages = []
+      while parser.has_next?
+        start_event = parser.pull
+        if start_event.start_element? and start_event[0] == 'message'
```
Comment (kou) [inline test/test_pullparser.rb:176]: Do we need `start_element?` check?

Can we simplify this like the following?

```ruby
_message_start = parser.pull
message_text = parser.pull
messages << message_text[0]
_message_end = parser.pull
parser.reset
```

**Comment by kou** [comment]
Thanks.

**Pull request merged**